src/fire.h

```diff
@@ -0,0 +1,26 @@
+#ifndef FIRE_H
+#define FIRE_H
+
+struct fire_data {
+    int fire_intensity;
+
+    int smoke_produced;
+    int fuel_produced;
+};
+
+struct mat_burn_data {
+    // Made to protect other materials too
+    bool immune;
```

Comment:
Defaults could go in the class rather than hidden in `item_factory`. The latter should probably use the `assign` syntax for consistency with the rest of that module.